### PR TITLE
#61 Implement render room list

### PR DESCRIPTION
**Pull request by TienNguyenDinh**
Ticket: [#61 Implement function to render rooms](https://trello.com/c/m0Mxd8Xr)

Preview
- Success
image

- Loading
image

- Error
image

### CODE REVIEW

File: hotel-management/src/pages/HomePage/index.tsx [hunk 1]
```diff
@@ -46,20 +22,15 @@ export const HomePage = () => {
...
-        ))}
-      </ul>
+      <div className="list-room-container">
+        {isLoading ? (
+          <Spinner size="lg" />
+        ) : error ? (
+          <Message content={{ FETCH_ROOMS_FAILED }} />
+        ) : (
+          <RoomList rooms={rooms} />
+        )}
```
Comment (hungdo-agilityio) [inline hotel-management/src/pages/HomePage/index.tsx]: Split into a function to render using switch case or if. This one is hard to read

File: hotel-management/src/pages/HomePage/index.tsx [hunk 2]
```diff
@@ -46,20 +22,15 @@ export const HomePage = () => {
...
-            onDelete={() => {}}
-            onEdit={() => {}}
-          />
-        ))}
-      </ul>
+      <div className="list-room-container">
+        {isLoading ? (
+          <Spinner size="lg" />
+        ) : error ? (
+          <Message content={{ FETCH_ROOMS_FAILED }} />
```
Comment (hungdo-agilityio) [inline hotel-management/src/pages/HomePage/index.tsx]: should this be `content={FETCH_ROOMS_FAILED}`

File: hotel-management/src/pages/HomePage/index.tsx [hunk 3]
```diff
@@ -1,41 +1,17 @@
 import { Link } from 'react-router-dom';
```
Comment (hungdo-agilityio) [inline hotel-management/src/pages/HomePage/index.tsx:1]: Have u tested UI when loading or error yet, and any image attach for this UI render?

Comment (TienNguyenDinh) [reply hotel-management/src/pages/HomePage/index.tsx:1]: Yes I found a minor bug on `useFetchRooms`, and the fix will be on another pr

File: hotel-management/src/pages/HomePage/index.tsx [hunk 4]
```diff
@@ -1,41 +1,27 @@
...
-      isAvailable: false,
-    },
-  ];
+  const { rooms, isLoading, error } = useFetchRooms();
+  const { FETCH_ROOMS_FAILED } = ROOM_MESSAGE;
+
+  const renderRoomList = () => {
+    if (isLoading) {
+      return <Spinner size="lg" />;
+    } else if (error) {
```
Comment (hungdo-agilityio) [inline hotel-management/src/pages/HomePage/index.tsx]: already return above, remove else statement

**Review by hungdo-agilityio — APPROVED** [review]

**Pull request merged**